src/v/utils/log_hist.h

```diff
@@ -185,7 +185,7 @@ class log_hist : public log_hist_base {
      */
     void record(uint64_t val) {
         _sample_sum += val;
-        const int i = std::clamp(
+        const unsigned i = std::clamp(
```

Comment:
nice, how did you arrive at this being the cause?

Reply:
The instruction was sign extending the register holding `i` into the register used to index the vector.  After I switched it to unsigned clang used the register storing `i` to index into the array. I guess the `unsigned` was enough to let clang know that `i` isn't going to be negative and won't need any sign extension when converting to a 64 bit index into the vector.

Reply:
l33t